src/main/java/frc/robot/commands/AutoChooser.java

```diff
@@ -0,0 +1,111 @@
+
+    //Robot starting positions
+    public static final double[] S0 = {94.5, 0};
+    public static final double[] S1 = {160.3, 0};
+    public static final double[] S2 = {234, 0};
+    public static final double[] S3 = {304.7, 0};
+
+    //Ball positions
+    public static final double[] B0 = {27.6, 122.6};
+    public static final double[] B1 = {27.6, 158.6};
+    public static final double[] B2 = {27.6, 194.6};
+    public static final double[] B3 = {214.3, 130.3};
+    public static final double[] B4 = {120.6, 115};
+    public static final double[] B5 = {145, 107.8};
+    public static final double[] B6 = {160.3, 114.2};
+    public static final double[] B7 = {175.6, 120.5};
+    public static final double[] B8 = {286.2, 130.4};
+    public static final double[] B9 = {304.7, 130.4};
+
+    //Robot ending positions
+    public static final double[] E1 = {94.5, -120};
+    public static final double[] E2 = {234, 146};
+
+    private AutoDrive autoDriver;
+    private Drivetrain m_drivetrain;
+
+    private double yaw;
+
+    public AutoChooser(Drivetrain m_drivetrain) {
+        this.autoDriver = new AutoDrive(m_drivetrain, 0, 0);
+        yaw = m_drivetrain.getYaw();
+    }
+    
+    //Returns the command associated with an enumerator
+    public CommandBase getCommand(AutoPath path, boolean dispense) {
+        switch (path) {
+            case QUICKSCORE:
+                return new QuickScore(m_drivetrain, dispense);
+            case TRENCHSCORE:
+                return new TrenchScore(m_drivetrain, dispense);
+            case GENERATORSCORE:
+                return new GeneratorScore(m_drivetrain, dispense);
+            case YEET:
+                return new Yeet(m_drivetrain, dispense);
+            case TRENCHSTEAL:
+                return new TrenchSteal(m_drivetrain, dispense);
+        }
+        return null;
+    }
+
+    //Auto path options
+    public enum AutoPath {
+        //Go straight to the lower port and dispense preloaded power cells
+        //Points: 11
+        QUICKSCORE,
+
+        //The previous, but moving to the trench first and collecting cells 0-1
+        //Points: 15
+        TRENCHSCORE,
+
+        //Move forward to the shield generator and collect 2 balls off the edge
+        //Points: 15
+        GENERATORSCORE,
+
+        //Move to the other side of the field to be in position to defend
+        //Points: 5
+        YEET,
+
+        //Steal balls 8-9 from the opponents control panel and score to the lower port
+        //Points: 15
+        TRENCHSTEAL;
+    }
+
+    //basic math/trig functions
+    public static double getAngle(double currentAngle, double[] pointA, double[] pointB) {
+        double delX = pointB[0]-pointA[0];
+        double delY = pointB[1]-pointA[1];
+
+        double angle;
+        if(delY > 0) {
+            angle = 90 - Math.atan(delY/delX) - currentAngle;
+        }
+        else {
+            angle = 0-90 - Math.atan(delY/delX) - currentAngle;
+        }
```

Comment:
This can be replaced by 
```
double angleRadians = Math.atan2(delY, delX);
double angle = angleRadians * 180. / Math.PI - currentAngle; // for degrees
```

See https://docs.oracle.com/javase/7/docs/api/java/lang/Math.html#atan2(double,%20double).

Keep in mind `atan` and `atan2` return radians.

Alternatively, another choice is to use the WPILib class for Rotation2d and Translation2d
```
Rotation2d getAngle(Rotation2d currentAngle, Translation2d pointA, Translation2d pointB) {
    Translation2d diff = pointB.minus(pointA);

    Rotation2d angle = new Rotation2d(diff.getX(), diff.getY()).minus(currentAngle);
    return angle;
}
```
https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/geometry/Rotation2d.html#%3Cinit%3E(double,double)
https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/geometry/Translation2d.html
https://github.com/wpilibsuite/allwpilib/blob/master/wpilibj/src/main/java/edu/wpi/first/wpilibj/geometry/Rotation2d.java